### PR TITLE
[SPARK-40540][SQL] Migrate compilation errors onto error classes: _LEGACY_ERROR_TEMP_1100-1199

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to migrate 100 compilation errors onto temporary error classes with the prefix `_LEGACY_ERROR_TEMP_11xx`. The error message will not include the error classes, so, in this way we will preserve the existing behaviour.

### Why are the changes needed?
The migration on temporary error classes allows to gather statistics about errors and detect most popular error classes. After that we could prioritise the work on migration.

The new error class name prefix `_LEGACY_ERROR_TEMP_` proposed here kind of marks the error as developer-facing, not user-facing. Developers can still get the error class programmatically via the `SparkThrowable` interface, so that they can build error infra with it. End users won't see the error class in the message. This allows us to do the error migration very quickly, and we can refine the error classes and mark them as user-facing later (naming them properly, adding tests, etc.).

### Does this PR introduce _any_ user-facing change?
No. The error messages should be almost the same by default.

### How was this patch tested?
By running the modified test suites:
```
$ build/sbt "core/testOnly *SparkThrowableSuite"
$ build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite"
$ build/sbt -Phive -Phive-thriftserver "test:testOnly *MetastoreDataSourcesSuite"
$ build/sbt -Phive -Phive-thriftserver "test:testOnly *BucketedWriteWithHiveSupportSuite"
$ build/sbt "test:testOnly org.apache.spark.sql.jdbc.JDBCWriteSuite"
$ build/sbt "test:testOnly *.PersistedViewTestSuite"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
@srielau @cloud-fan @anchovYu @gatorsmile Please, review this PR.

**Comment by MaxGekk** [comment]
cc @itholic

**Review by cloud-fan — APPROVED** [review]

**Comment by MaxGekk** [comment]
Only one test failed:
```
DAGSchedulerSuite.SPARK-40096: Send finalize events even if shuffle merger blocks indefinitely with registerMergeResults is false
org.scalatest.exceptions.TestFailedException: ArrayBuffer("hostB") was empty
```
but it is flaky, it seems, see https://github.com/apache/spark/pull/37989#discussion_r980652463.

Merging to master. Thank you, @cloud-fan for review.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 1]
```diff
@@ -1064,32 +1069,49 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
...
+    new AnalysisException(
+      errorClass = "_LEGACY_ERROR_TEMP_1103",
+      messageParameters = Map("clz" -> clz.toString))
   }
 
   def secondArgumentNotDoubleLiteralError(): Throwable = {
-    new AnalysisException("The second argument should be a double literal.")
+    new AnalysisException(
+      errorClass = "_LEGACY_ERROR_TEMP_1104",
+      messageParameters = Map.empty)
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:1098]: This appears quite a lot. Can we add a new constructor in `AnalysisException` which only takes an error class? @MaxGekk

Comment (MaxGekk) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:1098]: Added an item to the TODO task https://issues.apache.org/jira/browse/SPARK-39778

Comment (srielau) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:1098]: Aside from the fact that _LEGACY is temporary, isn't it "noteworthy" when an error takes no argument? That should be a small fraction.

Comment (srielau) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:1098]: @MaxGekk I'm proposing to leave the Map.empty in (status quo) as an explicit reminder in the code that this message takes no arguments

**Pull request closed without merging**